sdmetrics/reports/multi_table/_properties/inter_table_trends.py

```diff
@@ -0,0 +1,197 @@
+            'Parent Table',
+            'Child Table',
+            'Foreign Key',
+            'Column 1',
+            'Column 2',
+            'Metric',
+            'Score',
+            'Real Correlation',
+            'Synthetic Correlation',
+            'Error'
+        ])
+        for relationship in metadata['relationships']:
+            parent = relationship['parent_table_name']
+            child = relationship['child_table_name']
+            foreign_key = relationship['child_foreign_key']
+            primary_key = relationship['parent_primary_key']
+
+            real_parent = real_data[parent].add_prefix(f'{parent}.')
+            real_child = real_data[child].add_prefix(f'{child}.')
+            synthetic_parent = synthetic_data[parent].add_prefix(f'{parent}.')
+            synthetic_child = synthetic_data[child].add_prefix(f'{child}.')
+
+            child_index = f'{child}.{foreign_key}'
+            parent_index = f'{parent}.{primary_key}'
+
+            denormalized_real = real_child.merge(
+                real_parent, left_on=child_index, right_on=parent_index
+            )
+            denormalized_synthetic = synthetic_child.merge(
+                synthetic_parent, left_on=child_index, right_on=parent_index
+            )
+
+            child_meta = metadata['tables'][child]
+            parent_meta = metadata['tables'][parent]
+            merged_metadata = metadata['tables'][child].copy()
+            child_cols = {
+                f'{child}.{col}': col_meta for col, col_meta in child_meta['columns'].items()
+            }
+            parent_cols = {
+                f'{parent}.{col}': col_meta for col, col_meta in parent_meta['columns'].items()
+            }
+            merged_metadata['columns'] = {**child_cols, **parent_cols}
+            if 'primary_key' in merged_metadata:
+                merged_metadata['primary_key'] = f'{child}.{merged_metadata["primary_key"]}'
+
+            parent_child_pairs = itertools.product(parent_cols.keys(), child_cols.keys())
+
+            self._properties[(parent, child, foreign_key)] = SingleTableColumnPairTrends()
+            details = self._properties[(parent, child, foreign_key)]._generate_details(
+                denormalized_real, denormalized_synthetic, merged_metadata,
+                progress_bar=progress_bar, column_pairs=parent_child_pairs
+            )
+            details['Parent Table'] = parent
+            details['Child Table'] = child
+            details['Foreign Key'] = foreign_key
+            details['Column 1'] = details['Column 1'].str.replace(f'{parent}.', '', n=1)
+            details['Column 2'] = details['Column 2'].str.replace(f'{child}.', '', n=1)
+
+            all_details = pd.concat([all_details, details]).reset_index(drop=True)
+
+        if all_details['Error'].isna().all():
+            all_details = all_details.drop('Error', axis=1)
+        else:
+            all_details['Error'] = all_details['Error'].replace({np.nan: None})
```

Comment:
wondering if we can pull this out somehow since it seems like we do it in every property

Reply:
I ended up refactoring the base `.get_score` method to handle this. Now it will call `_generate_details` to create the details dataframe, then clean it up and return the average score.